Client/mods/deathmatch/logic/CClientGame.cpp

```diff
@@ -5544,6 +5544,10 @@ void CClientGame::ResetMapInfo()
     g_pGame->GetWeather()->ResetSandstorm();
     g_pGame->GetWeather()->ResetRainbow();
 
+    bool volumetricShadow;
+    g_pCore->GetCVars()->Get("volumetric_shadows", volumetricShadow);
+    g_pGame->GetSettings()->SetVolumetricShadowsEnabled(volumetricShadow);
```

Comment:
Maybe better to add a new method `SetVolumetricShadowsEnabledByScript` and just invoke it with `false` here?

Reply:
relying on the variable value not good m_bVolumetricShadowsEnabled will changes when u set VolumetricShadowes
Basically we will need to pass two parameters to indicate if it enable by checkbox or script
So I avoid this and just read the checkbox itself.

Reply:
Why do you need to change `m_bVolumetricShadowsEnabled`? You added `m_bVolumetricShadowsEnabledByScript` for this, isn't?

Reply:
I will add the function you mentioned SetVolumetricShadowsEnabledByScript to separate between function of checkbox and the one script fire

Reply:
I just add GetVolumetricShadowsEnabledByVideoSetting
and remove enable by script since I think it's useless 
see test
[https://youtu.be/E_Tl5meniEc?si=ERiqJeXRPrdpU2Xf](https://youtu.be/E_Tl5meniEc?si=ERiqJeXRPrdpU2Xf)
on reconnect it will set the video setting correctly